islands/filterpane/StampFilterPane.tsx

```diff
@@ -0,0 +1,501 @@
+      type="number"
+      value={value}
+      onChange={(e) => onChange(e.target.value)}
+      min="0"
+      className="w-full px-2 py-1 border border-purple-300 rounded-md focus:ring-2 focus:ring-purple-500 focus:border-purple-500"
+      placeholder="Enter value"
+    />
+  </div>
+);
+
+function useDebouncedCallback(callback: Function, delay: number) {
+  const timeoutRef = useRef(null);
+
+  function debouncedCallback(...args) {
+    clearTimeout(timeoutRef.current);
+    timeoutRef.current = setTimeout(() => {
+      callback?.(...args);
+    }, delay);
+  }
+
+  return debouncedCallback;
+}
+
+const defaultFilters = {
+  buyNow: {
+    atomic: false,
+    dispenser: false,
+  },
+  status: {
+    locked: false,
+    oneOfOne: false,
+  },
+  fileType: {
+    svg: false,
+    pixel: false,
+    gif: false,
+    html: false,
+    olga: false,
+    src721: false,
+    src101: false,
+  },
+  stampRange: {
+    min: "",
+    max: "",
+  },
+  priceRange: {
+    min: "",
+    max: "",
+  },
+  sortOrder: "descending",
+};
+
+export function filtersToQueryParams(
+  search: string,
+  filters: typeof defaultFilters,
+) {
+  const queryParams = new URLSearchParams(search);
+  Object.entries(filters).forEach(([category, value]) => {
+    if (typeof value === "object") {
+      Object.entries(value).forEach(([key, val]) => {
+        const strVal = val.toString();
+        console.log(key, strVal);
+        if (typeof val === "boolean") {
+          if (strVal !== "false") {
+            queryParams.append(`${category}[${key}]`, strVal);
+          }
+        } else {
+          queryParams.append(`${category}[${key}]`, strVal);
+        }
+      });
+    } else {
+      const strVal = value.toString();
+      if (typeof value === "boolean" && strVal !== "false") {
+        queryParams.append(category, value.toString());
+      }
+    }
+  });
+  return queryParams.toString();
+}
+
+function filtersToServicePayload(filters: typeof defaultFilters) {
+}
+
+export function queryParamsToFilters(query: string) {
+  const params = new URLSearchParams(query);
+  const filtersPartial = {};
+  Object.keys(defaultFilters).forEach((category) => {
+    if (category in defaultFilters) {
+      const filter = defaultFilters[category];
+      if (typeof filter === "object") {
+        Object.keys(filter).forEach((key) => {
+          const value = params.get(`${category}[${key}]`);
+          if (value !== null) {
+            if (!filtersPartial[category]) {
+              filtersPartial[category] = {};
+            }
+            filtersPartial[category][key] = value;
+          }
+        });
+      } else {
+        const value = params.get(category);
+        if (value !== null) {
+          filtersPartial[category] = value;
+        }
+      }
+    }
+  });
+
+  return {
+    ...defaultFilters,
+    ...filtersPartial,
+  };
+}
+
+export function queryParamsToServicePayload(query: string) {
+  return filtersToServicePayload(queryParamsToFilters(query));
```

Comment:
this will eventually be where we map the query params to the body the service is expecting to filter better.